### PR TITLE
Hide woo branding on partner-signup flow

### PR DESCRIPTION
**Pull request by lsl**
#### Proposed Changes

* Fixes 13520-gh-Automattic/woocommerce.com
* Replaces the WooCommerce branded oauth login screen when coming from the WordPress.com branded https://woocommerce.com/partner-signup/. 
* Other WooCommerce oauth logins should remain on the Woo branded screen.

#### Testing Instructions

* Visit https://woocommerce.com/partner-signup while logged out
* When redirected to wordpress.com, replace `https://wordpress.com` with `http://calypso.localhost:3000`
* The wordpress.com login screen should be shown instead of the woocommerce oauth screen.
* Repeat for https://woocommerce.com/vendor-signup, where the woo branded login screen should remain.

Alternatively, view the difference between these screens:
* http://calypso.localhost:3000/log-in
* Redirect from /partner-signup/ http://calypso.localhost:3000/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dsnip%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253Dpartner-signup%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26redirect_to%3Dhttps%3A%2F%2Fwoocommerce.com%2Fpartner-signup%2F
* Redirect from /vendor-signup/ http://calypso.localhost:3000/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3D%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253Dvendor-signup%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26redirect_to%3Dhttps%3A%2F%2Fwoocommerce.com%2Fvendor-signup%2F
* Reminder, if you get an access denied error at the end of the login flow its probably because you're using one of the above urls with a snipped access key, start the flow from logged out on both wpcom and woocom and visiting woo.com/partner-signup directly.
* Check link back to partners is localized along with login content http://calypso.localhost:3000/log-in/ja

Before
Screenshot 2022-06-07 at 16-50-54 Log In — WordPress com

After
Screenshot 2022-06-07 at 16-50-20 Log In — WordPress com

Fixes 13520-gh-Automattic/woocommerce.com

### CODE REVIEW

File: client/blocks/login/index.jsx [hunk 1]
```diff
@@ -240,57 +241,61 @@ class Login extends Component {
...
+												rel="noopener noreferrer"
+											/>
+										),
+										br: <br />,
+									},
+								}
+							) }
+						</p>
+					);
+				}
```
Comment (lsl) [inline client/blocks/login/index.jsx]: Did a bit of a reorg while adding the `isPartnerSignup` section.

Changed from 

```
if ( isWooOAuth2Client( oauth2Client ) && ! wccomFrom ) {
	//
}

if ( isWooOAuth2Client( oauth2Client ) && wccomFrom ) {
	//
}
```

To

```
if ( isWooOAuth2Client( oauth2Client ) ) {
	if ( isPartnerSignup ) {
		//
	} else if ( wccomFrom ) {
		//
	} else {
		//
	}
}
```

**Comment by lsl** [comment]
Can ignore e2e test failures for `specs/plans/plans__legacy-renew.ts: Plans (Legacy): Renew: Validate current plan (1)`, see p1654264833113049-slack-C1A1EKDGQ / https://github.com/Automattic/wp-calypso/issues/64395

**Comment by lsl** [comment]
Feel free to review as is, will add the [header from figma](13520-gh-Automattic/woocommerce.com#issuecomment-1148268853) tomorrow if it's not too much of an issue.

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (Rolilink) [reply client/blocks/login/index.jsx]: thanks for the clarification in here!

File: client/login/wp-login/login-links.jsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import config from '@automattic/calypso-config';
 import { getUrlParts } from '@automattic/calypso-url';
 import { Gridicon } from '@automattic/components';
 import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
```
Comment (Rolilink) [inline client/login/wp-login/login-links.jsx]: I think we are deprecating lodash per p4TIVU-9Bf-p2 can we find an alternative?

File: client/login/wp-login/login-links.jsx [hunk 3]
```diff
@@ -377,6 +381,7 @@ export default connect(
 		query: getCurrentQueryArguments( state ),
 		isJetpackWooCommerceFlow: 'woocommerce-onboarding' === getCurrentQueryArguments( state ).from,
 		wccomFrom: getCurrentQueryArguments( state )[ 'wccom-from' ],
+		isPartnerSignup: isPartnerSignupFlow( get( getCurrentQueryArguments( state ), 'redirect_to' ) ),
```
Comment (Rolilink) [inline client/login/wp-login/login-links.jsx]: maybe `getCurrentQueryArguments( state )?.redirect_to` could work here  to remove `get`

**Comment by Rolilink** [comment]
Works as intended

File: client/layout/masterbar/partner-signup.scss [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	&:visited, &:hover, &:focus {
+		color: var( --color-text );
+	}
+
+	.gridicon {
+		height: 20px;
+		margin-right: 2px;
+		top: 5px;
+		width: 20px;
+	}
```
Comment (lsl) [inline client/layout/masterbar/partner-signup.scss]: This gridicon could probably be cleaned up, just copied straight out of /start/domains.

File: client/state/login/utils.js [hunk 5]
```diff
@@ -176,3 +176,7 @@ export async function postLoginRequest( action, bodyObj ) {
 	}
 	throw new HTTPError( response, await response.text() );
 }
+
+export function isPartnerSignupFlow( redirectTo ) {
+	return redirectTo?.match( /woocommerce\.com\/partner-signup/ )?.length > 0;
+}
```
Comment (lsl) [inline client/state/login/utils.js:202]: Need to rewrite this as a hook and use the currentQuery hook in `client/components/route`, it temporarily fails when you click "create an account" because the url changes and double encodes the url (need to match `woocommerce\.com(\/|%2F)partner-signup`) as well as changing from `redirect_to` to a `oauth2_redirect` param.

File: client/layout/logged-out.jsx [hunk 6]
```diff
@@ -133,7 +139,10 @@ export default withCurrentRoute(
 		const sectionName = currentSection?.name ?? null;
 		const sectionTitle = currentSection?.title ?? '';
 		const isJetpackLogin = currentRoute.startsWith( '/log-in/jetpack' );
-		const isWhiteLogin = currentRoute.startsWith( '/log-in/new' );
+		const isPartnerSignup = isPartnerSignupQuery( currentQuery );
+		const isPartnerSignupStart = currentRoute.startsWith( '/start/wpcc' );
+		const isWhiteLogin =
+			currentRoute.startsWith( '/log-in/new' ) || ( isPartnerSignup && ! isPartnerSignupStart );
```
Comment (lsl) [inline client/layout/logged-out.jsx:152]: Not enjoying this hack on top of a hack setup but it's a required stepping stone if we want to fix the [flash of woo text](https://github.com/Automattic/wp-calypso/issues/64439) when transitioning to the user creation screen before completing the rebrand of that page. We'll be able to remove `isPartnerSignupStart` entirely in https://github.com/Automattic/wp-calypso/pull/64477.

File: client/state/login/utils.js [hunk 7]
```diff
@@ -176,3 +176,19 @@ export async function postLoginRequest( action, bodyObj ) {
 	}
 	throw new HTTPError( response, await response.text() );
 }
+
+export function isPartnerSignupQuery( currentQuery ) {
```
Comment (allilevine) [inline client/state/login/utils.js:186]: Would a code comment about what this does be helpful?

File: client/state/oauth2-clients/ui/selectors.js [hunk 8]
```diff
@@ -34,4 +34,6 @@ export const getCurrentOAuth2Client = ( state ) => {
  * @param  {object}   state  Global state tree
  * @returns {boolean}         Whether the OAuth2 layout should be used.
  */
-export const showOAuth2Layout = ( state ) => !! getCurrentOAuth2ClientId( state );
+export const showOAuth2Layout = ( state ) => {
+	return !! getCurrentOAuth2ClientId( state );
```
Comment (allilevine) [inline client/state/oauth2-clients/ui/selectors.js]: Why the change here?

File: client/layout/masterbar/partner-signup.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import { Gridicon } from '@automattic/components';
+import { useLocalizeUrl } from '@automattic/i18n-utils';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import WordPressLogo from 'calypso/components/wordpress-logo';
+
+import './partner-signup.scss';
+
+export default function PartnerSignupMasterbar() {
```
Comment (allilevine) [inline client/layout/masterbar/partner-signup.tsx]: I see why we need to create a different Masterbar from the design, but this seems like a lot of code for one link. Is that because of the hackiness? Is it worth doing?

**Review by allilevine — COMMENTED** [review]
@lsl This flow works as expected, but I see what you mean about it being hacky. It's like we're shoehorning in this other experience. At the same time, it's still the Woo oAuth experience -- just a different, partner flavor. I think we should see if we can get more opinions about alternative approaches, but trying a new oAuth client sounds good.

File: client/state/oauth2-clients/ui/selectors.js (hunk 8 above)
Comment (lsl) [reply client/state/oauth2-clients/ui/selectors.js]: I used this code earlier to do some partner page switching  but since removed, didn't revert the code.

File: client/layout/masterbar/partner-signup.tsx (hunk 9 above)
Comment (lsl) [reply client/layout/masterbar/partner-signup.tsx]: The (W) logo and actions setup in /log-in/new and /start reskin variants wasn't re-usable for this login flow, it made more sense to create something more straight forward. Tempted to just rename this SignupMasterbar or something and pass the link in.

File: client/components/wordpress-logo/index.jsx [hunk 10]
```diff
@@ -4,7 +4,7 @@ import './style.scss';
 
 function WordPressLogo( { className = 'wordpress-logo', size = 72 } ) {
 	return (
-		<svg className={ className } height={ size } width={ size } viewBox="0 0 72 72">
+		<svg className={ className } height={ size } width={ size } viewBox="0 -1 72 74">
```
Comment (lsl) [inline client/components/wordpress-logo/index.jsx]: (W) Logo was being clipped 1-2px.

Screenshot 2022-06-10 at 16-07-25 Create a site — WordPress com

File: client/login/wp-login/login-links.jsx [hunk 11]
```diff
@@ -126,7 +128,8 @@ export class LoginLinks extends Component {
 			isCrowdsignalOAuth2Client( this.props.oauth2Client ) ||
 			isJetpackCloudOAuth2Client( this.props.oauth2Client ) ||
 			this.props.isGutenboarding ||
-			this.props.isP2Login
+			this.props.isP2Login ||
+			this.props.isPartnerSignup
```
Comment (lsl) [inline client/login/wp-login/login-links.jsx:132]: Hides the "Return to woocommerce.com" link on the login screen.

File: client/layout/masterbar/partner-signup.tsx (hunk 9 above)
Comment (lsl) [reply client/layout/masterbar/partner-signup.tsx]: Renamed to MasterbarLogin, attempted to re-use `Masterbar` but it had too much baggage, cleaned the new component up a bit its quite a small component at this point. The goBackUrl prop should be moved to child elements using `Masterbar/Item`'s but I couldn't clear the styling on those buttons and opted for something simple for now. I think there needs to be a wider discussion about the masterbar+layout components and the GB reskin as it's not easy to integrate the two.

File: client/components/wordpress-logo/index.jsx (hunk 10 above)
Comment (lsl) [reply client/components/wordpress-logo/index.jsx]: Moved fix to its own PR https://github.com/Automattic/wp-calypso/pull/64517

**Comment by lsl** [comment]
Reverted, need to know which unit tests were failing.

**Comment by lsl** [comment]
Discovered bug with english link suggestion. Follow up to address here https://github.com/Automattic/wp-calypso/issues/64598

**Comment by paulopmt1** [comment]
We'll need to work on the 2fa screen too. As this isn't a blocker, we can split this part into another issue, so we can deploy what we have now:

image

I've [added the task here](https://github.com/Automattic/wp-calypso/issues/64634).

**Review by paulopmt1 — APPROVED** [review]
Good job @lsl! It's hard to add new behavior to this screen maintaining the code as we would like to in our first iteration.

Everything worked as expected, including the localized links. About the 2fa screen, if you want to make it in another task, sounds good to me, so we can deploy this one now.

**Review by Rolilink — APPROVED** [review]
LGTM!

**Comment by lsl** [comment]
> Everything worked as expected, including the localized links. About the 2fa screen, if you want to make it in another task, sounds good to me, so we can deploy this one now.

That's the right call for now, there's a few adding up, 13520-gh-Automattic/woocommerce.com#issuecomment-1155949972

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7378176

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7378176).
